waiter/src/waiter/auth/authentication.clj

```diff
@@ -23,15 +23,19 @@
 (def ^:const AUTH-COOKIE-NAME "x-waiter-auth")
 
 (defprotocol Authenticator
+  (process-callback [this request scheme operation]
+    "Process any requests that might come in after initiating authentication. e.g. receive a request
+    with an authentication assertion after redirecting a user to authenticate with an identity provider.")
   (wrap-auth-handler [this request-handler]
     "Attaches middleware that enables the application to perform authentication.
      The middleware should
-     - either issue a 401 challenge asking the client to authenticate itself,
+     - issue a 401 challenge, or redirect, to get the client to authenticate itself,
      - or upon successful authentication populate the request with :authorization/user and :authorization/principal"))
 
 (defn- add-cached-auth
-  [response password principal]
-  (cookie-support/add-encoded-cookie response password AUTH-COOKIE-NAME [principal (System/currentTimeMillis)] 1))
+  [response password principal & [age-in-seconds]]
+  (cookie-support/add-encoded-cookie response password AUTH-COOKIE-NAME [principal (clj-time.coerce/to-long (t/now))]
```

Comment:
In general, we should avoid using modules without requiring them in the `ns` section.
`clj-time.coerce` is aliased to `tc` in other places of the code.

Reply:
done